### DISCUSSION TITLE
Are there plans to get adv-fort working for the new premium releases anytime soon?

### DISCUSSION THREAD

**Discussion by JonathanLochridge** [DFHack/dfhack#4499] [Q&A] [answered]
Are there plans to get adc-fort working for the new premium releases anytime soon?

What are the main obstacles to getting it to work?
Or are you mostly waiting for everything to get solidified and it moved to stable?

Is the GUI going to need to be redone to work within the graphical context or anything?

Or are there any known incompatibilities with how the new version has been stuctured?

Or is it just a to-do nobody has gotten too yet?

It looks like the "mode" command is also not working.
Which removes my B-plan of just switching modes when needed.

Would getting mode working again with the new version be easier than adv-fort?

**Comment by myk002** [comment]
@warmist do you wanna take this one : )

re: `mode`, we've looked into it, and dynamically switching modes between fort and adventure is likely no longer viable, at least not with the simple "set the mode variable" technique that `mode` used. `gui/sandbox` successfully switches modes into Arena, but that is a much more controlled environment, and there is a fair amount of setup `gui/sandbox` does to make that switch happen. Even more setup would have to be written for adventure.

**Reply by JonathanLochridge** [reply]
So, If I understand correctly: Within fort mode you can enable arena-like controls on that particular map?
Using gui/sandbox. But would be unable to switch back.
And unable to switch to actual adventure mode.
And in adventure mode, the opposite is nor feasible either. (or switching to fort mode?)


I guess with enough "setup" maybe, but it would probably be very hard? and most likely no.

**Comment by warmist** [comment, marked as answer]
So adv-fort currently is in "life support" mode. I.e. i've done minimal changes so it would work and you can get it at [my fork](https://github.com/warmist/scripts/blob/advfort-quickfix/gui/advfort.lua). As for plans short term i will probably fix advfort some more to e.g. display errors better (not log into console). However for long term the script as it stands is very low quality both from code and ui experience. Script stood still while dfhack moved on both with what is possible and what is "normal". I think i'll start a new issue for development ideas/feedback.

**Reply by warmist** [reply]
I've started an issue for future adv-fort. [here](https://github.com/DFHack/dfhack/issues/4502)

**Reply by JonathanLochridge** [reply]
Oph, it's a shame to learn it's in such bad shape. Honestly those two features were my favourite parts of DF-Hack.

I did on occasion use something else, but it was like 90% of what I actually used in it.
Well I did use changing a units race/creature type too sometimes.

At least it seems adv-fort is feasible to get working and better again but there is a lot of work to do.

Oh, the embark assistant is great too for fort mode games.

If it's within my skill level I would love to help. Not sure how much I can do though.
I know some C++ but am not that good with it.
And am pretty unfamiliar with lua.
